int-tests/src/test-integration/java/io/neow3j/protocol/Neow3jWriteIntegrationTest.java

```diff
@@ -212,4 +216,43 @@ public void testSendToAddress_TransactionSendAsset() throws IOException {
         assertNotNull(tx);
     }
 
+    @Test
+    public void cancelTransactionFailsBecauseTransactionWasAlreadyConfirmed() throws IOException {
+        TransactionSendToken param = new TransactionSendToken(IntegrationTestHelper.NEO_HASH, BigInteger.ONE,
+                defaultAccountAddress());
+        Transaction tx = getNeow3j().sendToAddress(param).send().getSendToAddress();
+
+        waitUntilTransactionIsExecuted(tx.getHash(), getNeow3j());
+
+        NeoCancelTransaction cancel = getNeow3j()
+                .cancelTransaction(tx.getHash(), asList(Hash160.fromAddress(defaultAccountAddress())), null)
+                .send();
+
+        assertThat(cancel.getError().getMessage(), is("This tx is already confirmed, can't be cancelled."));
+    }
+
+    /*
+     * This test can have timing issues. The transaction is sent and then immediately cancelled. But, if the
+     * transaction is included in a block before the cancellation is processed, the test will fail.
+     */
+    @Test
+    public void cancelTransactionSucceedsWithSingleSigner() throws IOException {
+        TransactionSendToken param = new TransactionSendToken(IntegrationTestHelper.NEO_HASH, BigInteger.ONE,
+                defaultAccountAddress());
+        Transaction tx = getNeow3j().sendToAddress(param).send().getSendToAddress();
+
+        NeoCancelTransaction cancel = getNeow3j()
+                .cancelTransaction(tx.getHash(), asList(Hash160.fromAddress(defaultAccountAddress())), null)
+                .send();
+
+        assertFalse(cancel.hasError());
+        Transaction txOrCtx = cancel.getTransaction();
```

Comment:
Can you enlighten me what the context (i.e., `...OrCtx`) would include? Would that just be the original transaction in case it has already been processed?

Reply:
This has to do with [this issue](https://github.com/neow3j/neow3j/issues/1008). I'll explain it in our next meeting. But, here it was just a left over that should not be there. I corrected the naming.